### PR TITLE
Implement periodic gathering as a job in tech preview & latest fix

### PR DESCRIPTION
**Pull request by tremes**
This is a unrevert PR with fix for the missing `priorityClassName`
Please see the original PR https://github.com/openshift/insights-operator/pull/764

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [X] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

no new data

## Documentation

- `path/to/documentation.md`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes/No

## References

https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

**Comment by tremes** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-e2e-aws-sdn-techpreview

**Comment by tremes** [comment]
/payload-job periodic-ci-openshift-release-master-ci-4.14-e2e-gcp-sdn-techpreview

File: .golangci.yml [hunk 1]
```diff
@@ -66,7 +66,7 @@ linters:
   enable:
     - bodyclose
     - deadcode
-    - depguard
+    #- depguard
```
Comment (rluders) [inline .golangci.yml:69]: I'm already cleaning it up at #776, but I guess its fine. I'll have to rebase it anyway.

**Review by rluders — APPROVED** [review]

**Pull request merged**